### ISSUE TITLE
Quotes Refuses to Honor es6 Tempaltes

### ISSUE THREAD

**Issue by joshrhoades** [eslint/eslint#7273]
Ok, trolled through similar issues, and they all seem to end up being solved with the addition of `allowTemplateLiterals` rule for `quotes`, but that isn't working for me and I think I'm about out of options or missing something so blatantly obvious that I will literally mock a kangaroo just to cause him to punch me, so here we are typing things into the webernets as a last-ditch effort.

It seems as perhaps there is some Bermudian Traingulan (what, yeah I said it) thing where a confluence of rules are combined to make one of the pieces tip over. That or I am missing something ridiculous despite my efforts, or perhaps there is a true issue lurking.

The code itself is fine, it executes properly with no errors, nothing is off in that sense.

This happens with `single` or `double`, and whether or not I include the props for `allowEscape` or `allowTemplateLiterals` in any combination, including omitting them.

If I run only the `recommended` rules, no such errors are reported, though that shouldn't be surprising.

**Tell us about your environment**

* **ESLint Version:** @3.6.0 (via `rollup-plugin-eslint`@3.0.0)
* **Node Version:** @5.6.0
* **npm Version:** @3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:** `eslintrc.json`
```json
{
    "env": {
        "browser": true,
        "es6": true,
        "worker": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 6
    },
    "rules": {
        "indent": [2,"tab"],
        "eqeqeq": 2,
        "no-extra-boolean-cast": 0,
        "linebreak-style": [2, "unix"],
        "quotes": [2, "single", {
        	"avoidEscape": true,
        	"allowTemplateLiterals": true
        }],
        "block-scoped-var": 2,
        "prefer-template": 1,
        "no-extra-semi": 2,
        "semi": 2,
        "semi-spacing": 2,
        "no-unexpected-multiline": 2,
        "no-useless-concat": 2,
        "no-unused-vars": 2,
        "no-eval": 2,
        "no-dupe-class-members": 2,
        "no-unused-labels": 2,
        "valid-typeof": [2, {
        	"requireStringLiterals": true
        }],
        "use-isnan": 2,
        "no-template-curly-in-string": 2,
        "camelcase": [2, {
        	"properties": "always"
        }],
        "comma-style": [2, "first", {
        	"exceptions": {
	        	"ArrayExpression": true,
	        	"ObjectExpression": true
	        }
	    }],
	    "no-console": 2,
	    "no-debugger": 2,
	    "brace-style": 2,
	    "dot-notation": 2,
        "curly": [2, "all"],
	    "consistent-this": [2, "self"],
        "no-mixed-spaces-and-tabs": [2, "smart-tabs"],
        "no-empty": [2, {
        	"allowEmptyCatch": true
        }],
        "one-var": [2, "always"],
        "valid-jsdoc": [2, {
        	"requireParamDescription": true,
        	"requireReturnDescription": true,
        	"matchDescription": ".+",
        	"requireReturnType": true,
        	"preferType": { "Boolean": "boolean", "Number": "number", "Object": "object", "String": "string", "Function": "function", "Date": "date" }
        }]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
var someOtherVars
//L113
				, sFileName = options.filename || `export_${new Date().getTime()}.csv`
//L166
					oFrameDocument.body.innerHTML = `\r\n${sTextContent}\r\n`;
```

Additionally, in another file, this perfectly valid template string:
```js
			data._subscription = `${_CHANNEL}.${target}.${data._breed}.${data._breedName}`;
```
Reports this linter error:
```
  155:25  warning  Unexpected string concatenation  prefer-template
  155:36  error    Strings must use singlequote     quotes
  155:51  error    Strings must use singlequote     quotes
  155:73  error    Strings must use singlequote     quotes
```

**What did you expect to happen?**
Shouldn't have seen linter errors.

**What actually happened? Please include the actual, raw output from ESLint.**
No other errors are reported for these lines or files.
```
  113:40  warning  Unexpected string concatenation  prefer-template
  113:40  error    Strings must use singlequote     quotes
  113:77  error    Strings must use singlequote     quotes
  166:38  warning  Unexpected string concatenation  prefer-template
  166:38  error    Strings must use singlequote     quotes
  166:63  error    Strings must use singlequote     quotes
```

**Comment by kaicataldo** [comment]
I'm having trouble reproducing this locally. I ran the current master branch of eslint with the exact .eslintrc you provided on the following code:

```js
const data = {}
  , _CHANNEL = ''
  , target = '';

data._subscription = `${_CHANNEL}.${target}.${data._breed}.${data._breedName}`;
```
and I'm not seeing any errors. 

What happens if you run ESLint without rollup-plugin-eslint? Maybe a silly question, but is it running ESLint on the transpiled output? Just asking because I'd expect template literals to be transpiled to concatenated strings.

**Comment by joshrhoades** [comment]
@kaicataldo I think you're onto something. Late last night, after I posted this, I also noticed eslint complaining about `this$$1` in some cases when I was using `consistent-this`, which could only exist in the transpiled version. I also ran against pure eslint (no rollup or rollup-plugin-eslint) involved with the same config and did not get any issues with `quotes` or `prefer-template`.

So looks like this is an issue with the way rollup is either configured or triggers the linter to run on transpiled code, not eslint itself.

Sorry for the false alarm, but thanks for the help (I should have ran against a native eslint too before posting this).

**Comment by kaicataldo** [comment]
No worries, glad we figured it out! Hope you get the rollup issue figured out :)
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!

**Comment by vitorbal** [comment]
Thanks for the update, @joshrhoades!
Closing this, as the issue seems to be resolved. Please reopen if there is more to discuss.
